merge_from_mkdocs_material.py

```diff
@@ -0,0 +1,278 @@
+#!/usr/bin/env python3
+
+"""Applies changes from mkdocs-material to the current directory.
+
+The upstream mkdocs-material revision on which the current directory is based is
+assumed to be stored in the MKDOCS_MATERIAL_MERGE_BASE file.
+
+This script computes the differences between that revision and the current
+master branch, and then applies them to the current directory via 3-way merge.
+If there are conflicts, conflict markers are left in modified files.
+"""
+
+import argparse
+import contextlib
+import json
+import os
+import pathlib
+import shutil
+import subprocess
+import tempfile
+
+MKDOCS_EXCLUDE_PATTERNS = [
+    # mkdocs-specific configuration files
+    '.gitignore',
+    '.gitattributes',
+    '.github',
+    '.browserslistrc',
+    '.dockerignore',
+    'requirements.txt',
+    'setup.py',
+    'Dockerfile',
+    'MANIFEST.in',
+
+    # Generated files
+    'material',
+
+    # mkdocs-specific files
+    'src/*.py',
+    'src/mkdocs_theme.yml',
+    'src/404.html',
+    'mkdocs.yml',
+
+    # Unneeded files
+    'typings/lunr',
+    'src/assets/javascripts/browser/worker',
+    'src/assets/javascripts/integrations/search/worker',
+
+    # Files specific to mkdocs' own documentation
+    'src/overrides',
+    'src/assets/images/favicon.png',
+    'src/.icons/logo.*',
+    'docs',
+    'LICENSE',
+    'CHANGELOG',
+    'package-lock.json',
+    '*.md',
+]
+
+ap = argparse.ArgumentParser()
+ap.add_argument('--clone-dir', type=str, default='/tmp/mkdocs-material'),
+ap.add_argument('--patch-output',
+                type=str,
+                default='/tmp/mkdocs-material-patch')
```

Comment:
did you forget to add a help description? Or was it too complex for a single sentence?

Reply:
I added a help description.

Reply:
I really like the idea for this script. It will help others if they decide to fork your work...

Reply:
This script was previously here (https://github.com/google/tensorstore/blob/94be4f2e8715511bb60fc0a0eaf07335881673b3/docs/tensorstore_sphinx_material/merge_from_mkdocs_material.py#L1) but I forgot to add it to this repository previously.  I made a few improvements though.  To be clear it still requires a lot of manual work to upgrade.